pkg/controlplane/instance.go

```diff
@@ -493,6 +495,29 @@ func (c completedConfig) New(delegationTarget genericapiserver.DelegationTarget)
 		return nil
 	})
 
+	if PeerProxyFeatureEnabled() {
+		peerEndpointReconcilerCtrl := controlplane.New(controlplane.Config{
+			PublicIP: c.GenericConfig.PublicAddress,
+
+			PeerEndpointReconciler: c.GenericConfig.PeerEndpointReconciler,
+			EndpointInterval:       c.ExtraConfig.EndpointReconcilerConfig.Interval,
+
+			ServerId:          c.GenericConfig.APIServerID,
+			PublicServicePort: publicServicePort,
+			PeerBindAddress:   c.GenericConfig.PeerBindAddress,
+		}, clientset)
+		m.GenericAPIServer.AddPostStartHookOrDie("peer-endpoint-reconciler-controller",
+			func(hookContext genericapiserver.PostStartHookContext) error {
+				peerEndpointReconcilerCtrl.Start(hookContext.StopCh)
+				return nil
+			})
+		m.GenericAPIServer.AddPreShutdownHookOrDie("peer-endpoint-reconciler-controller",
+			func() error {
+				peerEndpointReconcilerCtrl.Stop()
+				return nil
+			})
+	}
```

Comment:
we should only need to pass to this controller this parameters `(identity string, peeraddress string)` , same as to the Lease

Reply:
done. Also need the EndpointInterval.